pkg/storage/allocator.go

```diff
@@ -491,6 +516,37 @@ func (a Allocator) RebalanceTarget(
 	if target == nil {
 		return nil, ""
 	}
+	// We could make a simulation here to verify whether we'll remove the target we'll rebalance to.
+	for len(candidates) >= 0 {
```

Comment:
Sorry for not noticing this last time, but `for len(candidates) >= 0` is a pretty meaningless loop condition. Did you mean `for len(candidates) > 0`? Or to just make this an infinite loop?

Reply:
Yeah, I just thought it will break in the loop condition. I have changed it as your suggestion.